event-sourcing/event-source-api/src/main/java/uk/gov/justice/services/eventsourcing/source/api/service/EventStreamPageService.java

```diff
@@ -0,0 +1,152 @@
+package uk.gov.justice.services.eventsourcing.source.api.service;
+
+import static uk.gov.justice.services.eventsourcing.repository.jdbc.Direction.BACKWARD;
+import static uk.gov.justice.services.eventsourcing.repository.jdbc.Direction.FORWARD;
+import static uk.gov.justice.services.eventsourcing.repository.jdbc.FixedPosition.HEAD;
+import static uk.gov.justice.services.eventsourcing.repository.jdbc.Position.first;
+import static uk.gov.justice.services.eventsourcing.repository.jdbc.Position.head;
+import static uk.gov.justice.services.eventsourcing.repository.jdbc.Position.positionOf;
+
+import uk.gov.justice.services.eventsourcing.repository.jdbc.Direction;
+import uk.gov.justice.services.eventsourcing.repository.jdbc.FixedPosition;
+import uk.gov.justice.services.eventsourcing.repository.jdbc.Position;
+import uk.gov.justice.services.eventsourcing.source.api.feed.common.Page;
+import uk.gov.justice.services.eventsourcing.source.api.feed.common.PagingLinks;
+import uk.gov.justice.services.eventsourcing.source.api.feed.event.EventStreamEntry;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+import javax.ws.rs.core.UriInfo;
+
+@ApplicationScoped
+public class EventStreamPageService {
+
+    private static final int PATH_SEGMENT_EVENT_STREAMS = 0;
+
+    private static final long ZERO = 0L;
+
+    @Inject
+    private EventStreamService eventStreamService;
+
+    public Page<EventStreamEntry> pageOfEventStream(final String position,
+                                                    final Direction direction,
+                                                    final long pageSize,
+                                                    final UriInfo uriInfo
+    ) throws SQLException, MalformedURLException {
+
+        final Position positionOf = positionOf(position);
+
+        final List<EventStreamEntry> eventStreamEntries = eventStreamService.eventStream(positionOf, direction, pageSize);
+
+        return pageOfEventStreamWithLinks(positionOf, pageSize, uriInfo, convertStreamIdToSelfLinks(eventStreamEntries, pageSize, uriInfo));
+    }
+
+    private List<EventStreamEntry> convertStreamIdToSelfLinks(final List<EventStreamEntry> eventStreamEntries,
+                                                              final long pageSize,
+                                                              final UriInfo uriInfo) throws MalformedURLException {
+        final List<EventStreamEntry> entriesWithSelfLinks = new ArrayList<>();
+
+        for (EventStreamEntry eventStreamEntry : eventStreamEntries) {
+            final Optional<URL> selfLink = selfLink(eventStreamEntry.getStreamId(), pageSize, uriInfo);
```

Comment:
is this optional? we are always doing .get() without checking?

Reply:
If there is an entry that will surely have a self link hence no checking when getting

Reply:
Ignore previous comment changed to not optional as If there is an EventStreamEntry that will surely have a steamId and a  link will be created